app/metrics/iq_activity.py

```
@@ -0,0 +1,208 @@
+        f.mkdir(parents=True, exist_ok=True)
+        return f
+
+    def detector_file(self, detector_id: str, name: str) -> Path:
+        """Get the path to a file which is used to track something specific to a detector."""
+        return Path(self.detector_folder(detector_id), name)
+
+    def increment_counter_file(self, name: str, detector_id: str | None = None):
+        """Increment a counter file, or create it if it doesn't exist. If detector_id is provided,
+        use the counter for that detector. Otherwise, use a system-wide counter.
+
+        Args:
+            name (str): The name of the counter file.
+            detector_id (str | None): The ID of the detector to use, if the counter is for a
+                specific detector. If None, use a system-wide counter.
+        """
+        if detector_id:
+            f = self.detector_file(detector_id, name)
+        else:
+            f = self.file(name)
+
+        if not f.exists():
+            f.touch()
+            f.write_text("1")
+            return
+
+        read_total = int(f.read_text())
+        f.write_text(str(read_total + 1))
+
+    def get_last_file_activity(self, name: str) -> datetime | None:
+        """Get the last time a file was modified."""
+        f = self.file(name)
+        if not f.exists():
+            return None
+        return datetime.fromtimestamp(f.stat().st_mtime)
+
+
+@lru_cache(maxsize=1)  # Singleton
+def _tracker() -> FilesystemActivityTrackingHelper:
+    """Get the activity tracker."""
+    return FilesystemActivityTrackingHelper(base_dir="/opt/groundlight/device/edge-metrics")
+
+
+def record_activity(detector_id: str, activity_type: str):
+    """Records an activity from a detector. Currently supported activity types are:
+    - iqs
+    - escalations
+    - audits
+    """
+    supported_activity_types = ["iqs", "escalations", "audits"]
+    if activity_type not in supported_activity_types:
+        logger.error(
```

I think I'd raise an exception here.  The only way this happens is through a coding bug / typo.  Logging an error makes it likely somebody doesn't notice, and the system continues along half-working.  For production systems, it's better to fail loudly and for the engineer to fix it instead of keep a half-working system online.  Especially if we know users can't trigger the error.